test/hpke_test.c

```diff
@@ -0,0 +1,1432 @@
+    size_t expected_secretlen;
+} TEST_EXPORTDATA;
+
+static int do_testhpke(const TEST_BASEDATA *base,
+                       const TEST_AEADDATA *aead, size_t aeadsz,
+                       const TEST_EXPORTDATA *export, size_t exportsz)
+{
+    OSSL_LIB_CTX *libctx = NULL;
+    const char *propq = testpropq;
+    OSSL_HPKE_CTX *sealctx = NULL, *openctx = NULL;
+    unsigned char ct[256];
+    unsigned char enc[256];
+    unsigned char ptout[256];
+    size_t ptoutlen = sizeof(ptout);
+    size_t enclen = sizeof(enc);
+    size_t ctlen = sizeof(ct);
+    unsigned char pub[OSSL_HPKE_TSTSIZE];
+    size_t publen = sizeof(pub);
+    EVP_PKEY *privE = NULL;
+    unsigned char authpub[OSSL_HPKE_TSTSIZE];
+    size_t authpublen = sizeof(authpub);
+    EVP_PKEY *authpriv = NULL;
+    unsigned char rpub[OSSL_HPKE_TSTSIZE];
+    size_t rpublen = sizeof(pub);
+    EVP_PKEY *privR = NULL;
+    int ret = 0;
+    size_t i;
+    uint64_t lastseq = 0;
+
+    if (!TEST_true(OSSL_HPKE_keygen(base->suite, pub, &publen, &privE,
+                                    base->ikmE, base->ikmElen, libctx, propq)))
+        goto end;
+    if (!TEST_true(cmpkey(privE, base->expected_pkEm, base->expected_pkEmlen)))
+        goto end;
+    if (!TEST_ptr(sealctx = OSSL_HPKE_CTX_new(base->mode, base->suite,
+                                              libctx, propq)))
+        goto end;
+    if (!TEST_true(OSSL_HPKE_CTX_set1_ikme(sealctx, base->ikmE, base->ikmElen)))
+        goto end;
+    if (base->mode == OSSL_HPKE_MODE_AUTH
+        || base->mode == OSSL_HPKE_MODE_PSKAUTH) {
+        if (!TEST_true(base->ikmAuth != NULL && base->ikmAuthlen > 0))
+            goto end;
+        if (!TEST_true(OSSL_HPKE_keygen(base->suite,
+                                        authpub, &authpublen, &authpriv,
+                                        base->ikmAuth, base->ikmAuthlen,
+                                        libctx, propq)))
+            goto end;
+        if (!TEST_true(OSSL_HPKE_CTX_set1_authpriv(sealctx, authpriv)))
+            goto end;
+    }
+    if (!TEST_true(OSSL_HPKE_keygen(base->suite, rpub, &rpublen, &privR,
+                                    base->ikmR, base->ikmRlen, libctx, propq)))
+        goto end;
+    if (!TEST_true(cmpkey(privR, base->expected_pkRm, base->expected_pkRmlen)))
+        goto end;
+    if (base->mode == OSSL_HPKE_MODE_PSK
+        || base->mode == OSSL_HPKE_MODE_PSKAUTH) {
+        if (!TEST_true(OSSL_HPKE_CTX_set1_psk(sealctx, base->pskid,
+                                              base->psk, base->psklen)))
+            goto end;
+    }
+    if (!TEST_true(OSSL_HPKE_encap(sealctx, enc, &enclen,
+                                   rpub, rpublen,
+                                   base->ksinfo, base->ksinfolen)))
+        goto end;
+    if (!TEST_true(cmpkey(privE, enc, enclen)))
+        goto end;
+    for (i = 0; i < aeadsz; ++i) {
+        ctlen = sizeof(ct);
+        OPENSSL_cleanse(ct, ctlen);
```

Comment:
`OPENSSL_cleanse` seems a bit over-the-top and heavy weight for a test. `memset` should be sufficient.

Reply:
sure